### PR TITLE
[azeventhubs] Update to latest azblob SDK

### LINKED ISSUE #19842: Event Hubs: Missing exported method "NewBlobStoreFromSharedKey" for creating checkpointer
### Bug Report

- import path of package in question: `github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs/checkpoints`
- SDK version: `0.4.0`
- output of `go version`: `1.19.4`

### Description

When creating a checkpointer using Azure Blob Storage, the package `github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs/checkpoints` offers two methods:

- `NewBlobStore` -> Uses Azure AD
- `NewBlobStoreFromConnectionString` -> Supports Azure Storage connection strings

A missing feature is being able to create a new blob store checkpointer from a shared key (a pair of account name and key).

I see that in the internal package that `checkpoints` references, `github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs/internal/blob`, there is already a method [`NewContainerClientWithSharedKey`](https://github.dev/Azure/azure-sdk-for-go/blob/2a983c938d002d4f27b2d277fc352b636b4d8eb5/sdk/messaging/azeventhubs/internal/blob/zc_container_client.go#L53-L54) which does precisely what we're looking for (and it depends on `NewSharedKeyCredential` which is another method in the same internal package). However, seems that there's no exported "higher level" method that leverages `NewContainerClientWithSharedKey`

### PR DESCRIPTION
**Pull request by richardpark-msft**
azblobs is approaching GA and we can now remove our vendored copy of azblobs and use the proper package. Our code footprint will get smaller and it'll also solve #19842, where we weren't exporting all the possible authentication for blobs.

Fixes #19842 #19883

### CODE REVIEW

**Comment by richardpark-msft** [comment]
NOTE: this uses the pre-release commit for azblobs so we can get close to what we need. We'll want to wait to merge this and release until after the GA release for azblobs goes out.

File: sdk/messaging/azeventhubs/checkpoint_store.go [hunk 1]
```diff
@@ -31,9 +33,9 @@ type Ownership struct {
 	FullyQualifiedNamespace string
 	PartitionID             string
 
-	OwnerID          string    // the owner ID of the Processor
-	LastModifiedTime time.Time // used when calculating if ownership has expired
-	ETag             string    // the ETag, used when attempting to claim or update ownership of a partition.
+	OwnerID          string      // the owner ID of the Processor
+	LastModifiedTime time.Time   // used when calculating if ownership has expired
+	ETag             azcore.ETag // the ETag, used when attempting to claim or update ownership of a partition.
```
Comment (jhendrixMSFT) [inline sdk/messaging/azeventhubs/checkpoint_store.go:38]: Does this need to be `*azcore.ETag`?  I know we use that in other places as I believe that an empty-string ETag is semantically different from no ETag.

Comment (richardpark-msft) [reply sdk/messaging/azeventhubs/checkpoint_store.go:38]: Ah, interesting. I'm good with being explicit, I'll switch it over.

Comment (jhendrixMSFT) [reply sdk/messaging/azeventhubs/checkpoint_store.go:38]: You might want to double-check how this is used in other places, but I'm pretty sure you'll want this change.

Comment (richardpark-msft) [reply sdk/messaging/azeventhubs/checkpoint_store.go:38]: Swapped over, seemed clean enough. I don't intend to interpret this value, so I think it's best to just completely stay out of that game on the client.

**Comment by richardpark-msft** [comment]
Interesting test failure - it's in an area completely untouched by my change. Going to retry.

File: sdk/messaging/azeventhubs/go.mod [hunk 2]
```diff
@@ -3,10 +3,11 @@ module github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs
 go 1.18
 
 require (
-	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.1.0
+	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.3.0
 	github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.1.0
 	github.com/Azure/azure-sdk-for-go/sdk/internal v1.1.2
 	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/eventhub/armeventhub v1.0.0
+	github.com/Azure/azure-sdk-for-go/sdk/storage/azblob v0.6.2-0.20230201191316-b8690e0d2585
```
Comment (jhendrixMSFT) [inline sdk/messaging/azeventhubs/go.mod:10]: v1.0.0 is now available.

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**